### PR TITLE
Appending path of userdef_environments to sys.path

### PR DESCRIPTION
**Pull request by dragly**
For some reason the current path is no longer present in sys.path when doconce.py tries to import userdef_environments.py. This patch includes the current path temporarily during the import. This issue may however be fixed by figuring out why the current path (also known as '') is removed from sys.path in the first place.

Note that this issue occured after installing doconce and all dependencies in a virtual environment. However, when running python in terminal, sys.path includes the current path ('') by default. This can be checked by running:

```
import sys
print sys.path
```

In other words, something seems to modify sys.path by mistake.

### CODE REVIEW

**Comment by hplgit** [comment]
Merged the fix, but not sure this is necessary. It is the script's path that is automatically added to `sys.path`, not the current working directory. When running `doconce` (located in some system `bin` directory), '' is not added to `sys.path`. However, the `import` statement should always be able to import from the current working directory. It does not need any '' in `sys.path`; it's just that if you run a script from the current working directory, that directory, as '', is added to `sys.path`.
